configs/components/ruby-2.7.8.rb

```diff
@@ -203,19 +207,21 @@
   rbconfig_changes = {}
   if platform.is_aix?
     rbconfig_changes["CC"] = "gcc"
-  elsif platform.is_cross_compiled? || platform.is_solaris?
+  elsif platform.is_macos? && platform.architecture == 'arm64'
     if platform.name =~ /osx-11/
```

Comment:
I think that these `osx-11` and `osx-12` conditions need to be moved down inside the `elsif platform.is_cross_compiled? || platform.is_solaris?` block.

Reply:
I think this is fine as the target will get set for 11 & 12 as well once they pass this condition. I will confirm once by running build for them as well on x86_64.

Reply:
It might be less confusing to do something like https://github.com/puppetlabs/puppet-runtime/pull/680/files#r1253820661

```ruby
elsif platform.is_macos?
  if platform.is_cross_compiled?
    rbconfig_changes["CC"] = 'clang -target arm64-apple-macos11' if platform.name =~ /osx-11/
    rbconfig_changes["CC"] = 'clang -target arm64-apple-macos12' if platform.name =~ /osx-12/
  else
    rbconfig_changes["CC"] = 'clang'
  end
elsif platform.is_solaris?
  ...
```